### DISCUSSION TITLE
Is there any way to make all windows in the current workspace get tabbed without manually moving them?

### DISCUSSION THREAD

**Discussion by moshpirit** [i3/i3#6081] [Q&A] [answered]
I find tab layout very useful and I use it quite often, but is uncomfortable to move the other windows to the focused one. Is there any way to do that automatically when pressing the keybind?

**Comment by okraits** [comment]
If changing the layout to tabbed doesn't affect all windows, it could mean that the current window is a split container. You need to use `focus parent` first to focus the toplevel container in the workspace and after that you can change the layout.

**Reply by moshpirit** [reply]
Hi, thanks for replying! This worked but it gave a strange result: https://i.imgur.com/z16Bhzn.png

**Comment by orestisfl** [comment, marked as answer]
Try
```
bindsym $mod+w layout tabbed, mark _tabbed, [workspace=__focused__] move to window mark _tabbed
```
for an all-in-one command

**Reply by moshpirit** [reply]
this made it! But it's strange that after doing it and going back to split layout, it shows the windows in a different layout than used before (with autotiling). So if I have two terminals open, they originally are shown side to side, but after tabbing and splitting them back, they are shown one on top of the other. But if I open more than 3, they are shown side to side. Is there any way to solve this?

**Reply by orestisfl** [reply]
This command explicitly moves windows away from their containers so no, it's not easy (without some scripting) to go back to the exact previous layout.

However,

> Is there any way to solve this?

can you perhaps be more explicit on what layout you would like to have instead when you go back to splits?

**Reply by moshpirit** [reply]
Oh, I just wanted to have the previous layout (auto-tiling). If it's not possible then it's fine as it is.

**Reply by orestisfl** [reply]
I am not really familiar with the auto-tiling repo. You could of course automatically re-arrange the windows using the IPC interface or save the layout and re-apply it but both of those are more involved.

Take a look here if you want to invest the time: https://i3wm.org/docs/ipc.html

**Reply by moshpirit** [reply]
Thanks for the info! :blush:

**Reply by moshpirit** [reply]
Hello, sorry for bringing this issue up again but I noticed that the titles of the windows are not correctly updated. Is there any way to 
![screenshot](https://i.imgur.com/8U67bil.png)
